src/Swarm/TUI/Controller.hs

```diff
@@ -516,3 +522,29 @@ handleInfoPanelEvent s (VtyEvent ev) = do
       let s' = s & uiState . uiInventory . _Just . _2 .~ l'
       continue s'
 handleInfoPanelEvent s _ = continueWithoutRedraw s
+
+-- | Handle user input events searching in the info panel.
+handleSearchingEvent :: AppState -> BrickEvent Name AppEvent -> EventM Name (Next AppState)
+handleSearchingEvent s (VtyEvent (V.EvKey V.KEsc []))
+  = continue $ s & uiState . uiSearching .~ False
+                 & uiState . uiSearch .~ ""
+                 & uiState . uiInventory . _Just . _1 .~ 0
+
+handleSearchingEvent s (VtyEvent (V.EvKey V.KEnter []))
+  = continue $ s & uiState . uiSearching .~ False
+                 & uiState . uiInventory . _Just . _1 .~ 0
```

Comment:
Should we remove the search on enter? otherwise it seems like we need to search again and press escape to clear the search.

Reply:
Alternatively, we could merge the KEsc handler with the global one, so that even when not in search mode, pressing Esc would still clear the search.

Reply:
I liked the idea of a "sticky" search, but perhaps that is not a wanted feature.  I agree it is awkward to have to enter search and hit <kbd>Esc</kbd> to clear it.